test/boost/sstable_3_x_test.cc

```diff
@@ -3023,20 +3026,18 @@ static flat_mutation_reader_v2 compacted_sstable_reader(test_env& env, schema_pt
     cf->mark_ready_for_writes();
     lw_shared_ptr<replica::memtable> mt = make_lw_shared<replica::memtable>(s);
 
-    tmpdir tmp;
     auto sstables = open_sstables(env, s, format("test/resource/sstables/3.x/uncompressed/{}", table_name), generations);
     auto new_generation = generations.back() + 1;
 
     auto desc = sstables::compaction_descriptor(std::move(sstables), default_priority_class());
-    desc.creator = [s, &tmp, &env, new_generation] (shard_id dummy) {
-        return env.make_sstable(s, tmp.path().string(), new_generation,
-                         sstables::sstable_version_types::mc, sstable::format_types::big, 4096);
+    desc.creator = [s, &env, new_generation] (shard_id dummy) {
+        return env.make_sstable(s, new_generation);
     };
     desc.replacer = replacer_fn_no_op();
     auto cdata = compaction_manager::create_compaction_data();
     sstables::compact_sstables(std::move(desc), cdata, cf->as_table_state()).get();
 
-    auto compacted_sst = open_sstable(env, s, tmp.path().string(), new_generation);
+    auto compacted_sst = env.reusable_sst(s, new_generation).get();
```

Comment:
open_sstable() opened the `mc` version, this defaults to highest-version which is `me`

Presumably there's more of that kind below, and I admit this switch was deliberate to let test-cases test newer versions of sstables, but in that case it should be at least reflected in the commit message